activerecord/test/cases/insert_all_test.rb

```diff
@@ -863,6 +863,32 @@ def test_insert_all_when_table_name_contains_database
     end
   end
 
+  def test_insert_all_with_unpersisted_records_deprecation
+    author = Author.create!(name: "Rafael")
+    author.books.build(title: "Unpersisted Book")
+    author.books.load
+
+    assert_deprecated(ActiveRecord.deprecator) do
+      author.books.insert_all([{ title: "New Book" }])
```

Comment:
Tests are missing for all other methods.

Reply:
Added the other pertinent methods